Migrate Documentation from wiki-pages [Archived: Refer#6]

vipulgupta2048 opened the pull request:
This PR focuses on the migration of wiki-pages of the activity " i-can-read " to its Github repository. Also demonstrating and testing how my script is working and what I wanted to improve more on. 

**Footnotes**
Used **WikiPort** Script Beta 2 and some manual table fixes.
- [x] Table support for converting to Markdown needs to be added to the script.

Reference - GSoC 2018 Project on Migration of wiki Activity pages to git [**WikiPort**]

Status - Complete

vipulgupta2048 committed:
Created Readme.md

Using WikiPort script Beta 2 and some manual table fixes.

rhl-bthr commented:
Reviewed.

Suggested Changes:
 - Kindly update the commit message; It should not include tags which are not be relevant to the change being proposed, in this case being `#GSoC2018`. The reason being these tags will be irrelevant for future developers while they might be going throught the git history
 - The links present in the "Contents" in the Readme point back to the wiki page. Kindly update them as well.
 - The links to the activities `AEIOU`, `Letter Match` and `I know my ABC's` will also have to be updated once their wiki pages are also migrated. Kindly add this to your checklist so that these links do not get broken once you migrate their wiki pages

vipulgupta2048 commented:
@Pro-Panda a very good review indeed. It seems I am a bit too excited about it. Removing. 
Thanks.

vipulgupta2048 commented:
@Pro-Panda @quozl  Changes you requested has been implemented. Here's a review. 
1. Changed git commit message. 
2. Fixed the links in the Content section.
3. The issue I am facing with the migration of `AEIOU`, `Letter Match` and `I know my ABC's` is that their code is not hosted on GitHub but on the Sugar Labs Gitorious Server. Solutions to this are as follows. 
- Migrate the codebase to [sugarlabs](www.github.com/sugarlabs)
- **or** Migrate the wiki to the Gitorious Server (This would help very little)

I am adding the migration of these 3 pages to my personal task list. I will migrate them after having a discussion. Till then this PR is ready to merge.

rhl-bthr commented:
Reviewed upto 6d33112

rhl-bthr commented:
@vipulgupta2048 

> The issue I am facing with the migration of AEIOU, Letter Match and I know my ABC's is that their code is not hosted on GitHub but on the Sugar Labs Gitorious Server

Github Links: [AEIOU](https://github.com/sugarlabs/AEIOU), [Letter Match](https://github.com/sugarlabs/lettermatch), [I know my ABC's](https://github.com/sugarlabs/iknowmyabcs)

There is a `search repositories` option on https://github.com/sugarlabs/ which you may use if you wish to search for repositories

> Migrate the codebase to sugarlabs

Your link to sugarlabs is broken


In my opinion, merging this now could lead to broken links which one might forget to fix later. Instead, while migrating any documentation, you may do the following:
 - Migrate the documentation from wiki to GitHub and submit a PR for the same
 - If there are any links which will get broken afterwards (probably links to other activities documentation), mark that PR as [In progress]
 - Migrate the pages in these links
 - Update the In progress PR and change its status from In-Progress to complete

quozl commented:
Thanks.  Reviewed to 6d33112.

* [x] images are linked to the Wiki,
* [x] there are links to Wiki activity pages,
* [x] there are links to a developer repository.

rhl-bthr commented on README.md:
```diff
@@ -0,0 +1,28 @@
+
+## I can read (Soy capaz de leer)
+
+I can read is an activity for introducing the basics of phonetic reading. It displays letters and images and associated sound files, such as 'A as in apple'. It also puts the letters into a context that coalesces into words and phrases as more letters are added. Students canttps://github.com/sugarlabs/iknowmyabcs hear sounds and record their own voice either repeating the sound or reading aloud. 
```
Kindly fix `canttps://github.com/sugarlabs/iknowmyabcs`

vipulgupta2048 commented:
@Pro-Panda  My bad, all bugs have been resolved. Please check now. Do check if all the suggestions that made by you have been implemented or not. Would make any changes if needed.

vipulgupta2048 commented:
This page contains an embedded audio file. My script didn't work for it, so should I try getting it manually through other methods. As I remember, markdown doesn't support features to embed audio. 
Also, implemented status in the PR description of every made.

quozl commented:
Your commit message is terse.  See [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits) for ideas of what to include.  Intended audience is a developer of the activity.

quozl commented:
Is `assets` a good name?  We most often use `images`.

quozl commented:
Your commit message is terse.  See [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits) for ideas of what to include.  Intended audience is a developer of the activity.

Is `assets` a good name?  We most often use `images`.

In your migration procedure, the pull request or commit message could also link to the Wiki page being migrated.  This would have saved your reviewer some time.

As this is to be an automated process, I think one commit should be used, rather than one commit for images and one for text.

Any invitations to the Talk page should be redirected to GitHub issues for the activity.

vipulgupta2048 commented:
> Your commit message is terse. See making commits for ideas of what to include. Intended audience is a developer of the activity.

I read the making commits section of contributing.md. Now, I want to amend my git messages. Would the example below prescribed as the right practice. 
Example - Migrated i-can-read activity as commit message
while the PR description will include the complete link of the page migrated.     

> Is assets a good name? We most often use images.

A good idea. But almost every repository had a images and screenshots folder. I didn't want to disturb anybody's else folder structure. Thus created a new one. Will Implement the same.

> In your migration procedure, the pull request or commit message could also link to the Wiki page being migrated. This would have saved your reviewer some time.

Yes, I should from now on. Thanks !! (Will mention the same in the PR's made)

> As this is to be an automated process, I think one commit should be used, rather than one commit for images and one for text.

Since this was my first PR to Sugar Labs, I used a beta version of the script which created only one commit. But then I realized that there were some components missing which I didn't account in my script. Hence the 2 commits. I have implemented those changes now. So it wouldn't be a problem in the future Pull Requests. Should I squash them for now?  

> Any invitations to the Talk page should be redirected to GitHub issues for the activity.

I had that doubt when I was going through the page, as my script did throw a 404. Will be adding the same.

quozl commented:
We're suffering from not collectively planning out the project idea.  @walterbender, mentioned just to ensure synchronisation.  @Aruwajiore @rdrsadhu @saichethan mentioned because they have been working on similar proposal.  This is the middle of a pull request, so please read context.

@vipulgupta2048 wrote:
> want to amend my git messages. Would the example below prescribed as the right practice.
> Example - Migrated i-can-read activity as commit message
> while the PR description will include the complete link of the page migrated.

* use _migrate_ rather than _migrated_, to fit the imperative mood in the checklist,
* the name of the activity is redundant; whoever is reading the commit should already know which repository they are looking at,
* the link of the page migrated should be in the commit message; it will remain useful in future, because of the Internet Archive or the Wiki history.

A worked example using [the checklist](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits);

* start with a one line summary of the change; _"Migrate documentation from Wiki"_,
* leave a blank line after the summary;
* explain the problem that is solved, unless the summary makes it obvious; _"Documentation from http://wiki.sugarlabs.org/go/Activities/I_can_read was separate from the activity."_,
* when the problem was introduced by a previous commit, mention the hash; _not relevant_,
* when the problem is in an issue or ticket, add "Fixes #1234"; _"Part of a project to migrate activity documentation from Wiki."_,
* avoid mentioning GitHub or other pull-requests, as these are not kept in git; _not relevant_,
* avoid mentioning any contest tasks or mentors; use pull-request comments instead;  _not relevant_; and
* use imperative mood, like "add foo", or "port to bar"; _use "Migrate" rather than "Migrated"_,

```
Migrate documentation from Wiki

Documentation from http://wiki.sugarlabs.org/go/Activities/I_can_read
was separate from the activity.

Part of a project to migrate activity documentation from Wiki.
$PROJECT_URL
```

@vipulgupta2048 wrote:
> A good idea. But almost every repository had a images and screenshots folder. I didn't want to disturb anybody's else folder structure. Thus created a new one. Will Implement the same.

Good point.  Don't want to disturb what the activity uses.

Please use `doc` or `docs`.  `doc` seems the most common, and is used by the filesystem hierarchy standard, and many GNU projects.

To avoid disturbing, code your script to check for the directory or output files first.

> Should I squash them for now?

Depends.  Are you planning another pull request or do you think this one should be merged?

vipulgupta2048 commented:
@Aruwajiore @rdrsadhu @Saichethan I think the points put forward by @quozl is imperative for the future development of the project. No matter what, we are all contributors to Sugar Labs. GSoC or not. 

Hence I invite you all to my [checklist](https://etherpad.net/p/Sugar_Migration_Checklist) that I created for the migration process that I implement. Feel free to make any changes in the format and I hope we can work together on this project for the time being. This will also act as a reference to whosoever would read this conversation later and help them too.

thecraftman commented:
@vipulgupta2048 alright.. Looking at it now
